js-ex2 3/cart.js

```diff
@@ -0,0 +1,64 @@
+var dataCart = JSON.parse(localStorage.getItem('cart'));
+console.log( dataCart);
+var $ItemCart = document.getElementById('js-card');
+var subTotal;
+var count;
+var idcart;
+function listCart() {
+  for (var i = 0, leng =  dataCart.length; i < leng; i++) {
+    idcart = i + 1;
+    count = parseInt( dataCart[i].quantity);
+    priceCart = parseInt( dataCart[i].product.price);
+    subTotal = count * priceCart;
+    $ItemCart.innerHTML += '<tr id ="jsRow' + idcart + '">' +
+      '<td>' + idcart + '</td>' +
+      '<td>' +  dataCart[i].product.titleProduct + '</td>' +
+      '<td><img class="img-cart-item" src="images/' +  dataCart[i].product.imgProduct + '"></td>' +
+      '<td class="td-quantity">' +  dataCart[i].quantity + '</td>' +
+      '<td> &#36;' +  dataCart[i].product.price + '<td>	&#36;' + count * priceCart + 
+      '</td></td><td><button class="js-btn-delete" id = "del-' +  dataCart[i].product.id+'">' +
+      // '</td></td><td><a onclick = "javascript:deleteCart(' +  dataCart[i].product.id + ')">' +
+      '<img class="img-delete" src="images/download-removebg-preview.png" alt=""></button></td>'
+      + '</tr>';
+  }
+  document.getElementById('js-total').innerHTML = '&#36;' + sumTotal();
+}
+
+function sumTotal() {
+  var sum = 0;
+  for (var j = 0, length =  dataCart.length; j < length; j++) {
+    sum += parseInt( dataCart[j].quantity) * parseInt( dataCart[j].product.price);
+  }
```

Comment:
I don't think you need `length = dataCart.length` here, just use `j < dataCart.length` instead.
Remember to check this one for other places also.